### DISCUSSION TITLE
Rolling shutter correction?

### DISCUSSION THREAD

**Discussion by Ken-Kuroki** [UTokyo-FieldPhenomics-Lab/EasyIDP#82] [Q&A] [answered]
When loading a PIX4D or Metashape project and running reverse projection to crop the original images, does EasyIDP take the rolling shutter effect (https://www.pix4d.com/blog/rolling-shutter-correction/) into account? The warping can distort the shape of the region to be cropped, so I'm curious if the result region is correct despite the effect.

In addition, even if the region is correct, the transformation is still needed to get proper images. Is there an easy way to extract the details of distortion calculated, so that I can implement correction transformation?

Thanks as always in advance!

**Comment by HowcanoeWang** [comment, marked as answer]
The EasyIDP does not handle such effect and uses the camera results directly (hard to know whether these results contains this effects or not).

When the flying speed is slow (in most agriculture applications) this effect does not affect too much. But instead, the EasyIDP takes the lens distortion into consideration (https://support.pix4d.com/hc/en-us/articles/202559069-Camera-distortion)

There is a parameter `distortion_correct ` of [Pix4D.back2raw()](https://easyidp.readthedocs.io/zh_CN/latest/python_api/autodoc/easyidp.pix4d.Pix4D.html#easyidp.pix4d.Pix4D.back2raw) controls whether do correction or not, by default, doing this correction.

(The reason why giving this handle, the Pix4D can export the distortion corrected images to a folder, turning it off to get coord on such corrected images.)

For Metashape project [Metashape.back2raw()](https://easyidp.readthedocs.io/zh_CN/latest/python_api/autodoc/easyidp.metashape.Metashape.html#easyidp.metashape.Metashape.back2raw), it does not providing such handle to control, forcing do this correction by default (because metashape can not export the distortion corrected images).

**Reply by Ken-Kuroki** [reply]
Thanks for detailed reply for both when using PIX4D and when using Metashape! I understand it now.
